components/controller.ts

```diff
@@ -479,6 +479,65 @@ export class Controller {
         this._addElusiveTargets()
         this.index()
 
+        if (modFrameworkDataPath && existsSync(modFrameworkDataPath)) {
+            log(
+                LogLevel.INFO,
+                "Simple Mod Framework installed - using the data it outputs.",
+            )
+
+            const lastServerSideData = (
+                parse(readFileSync(modFrameworkDataPath!).toString()) as {
+                    lastServerSideStates: {
+                        unlockables?: Unlockable[]
+                        contracts?: {
+                            [k: string]: MissionManifest
+                        }
+                        blobs?: Record<string, string>
+                    }
+                }
+            ).lastServerSideStates
```

Comment:
You might as well type this properly, a type may already exist for it as we read it for the deployed mods. If it does not, it should be refactored to use a type.

Reply:
Then see the latter part of my comment.